common/src/main/java/io/netty/util/Recycler.java

```diff
@@ -349,6 +358,20 @@ boolean transfer(Stack<?> dst) {
                 return false;
             }
         }
+
+        @Override
+        protected void finalize() throws Throwable {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Do not override the Object.finalize() method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AObjectFinalizeOverridenCheck)